pkg/orchestrator/assetscanestimationwatcher/watcher.go

```diff
@@ -0,0 +1,269 @@
+		FirstSeen:    assetScanEstimation.Asset.FirstSeen,
+		Id:           &assetScanEstimation.Asset.Id,
+		LastSeen:     assetScanEstimation.Asset.LastSeen,
+		Revision:     assetScanEstimation.Asset.Revision,
+		ScansCount:   assetScanEstimation.Asset.ScansCount,
+		Summary:      assetScanEstimation.Asset.Summary,
+		TerminatedOn: assetScanEstimation.Asset.TerminatedOn,
+	}
+
+	stats := w.getLatestAssetScanStats(ctx, asset)
+
+	estimation, err := w.provider.Estimate(ctx, stats, asset, assetScanEstimation.AssetScanTemplate)
+	if err != nil {
+		logger.Errorf("Failed to estimate asset scan: %v", err)
+		assetScanEstimation.State.State = utils.PointerTo(models.AssetScanEstimationStateStateFailed)
+		assetScanEstimation.State.Errors = utils.PointerTo(utils.UnwrapErrorStrings(err))
+	} else {
+		logger.Infof("Asset scan estimation completed successfully. Estimation=%v", estimation)
+		assetScanEstimation.State.State = utils.PointerTo(models.AssetScanEstimationStateStateDone)
+		assetScanEstimation.Estimation = &models.Estimation{
+			Cost:          estimation.Cost,
+			Size:          estimation.Size,
+			Time:          estimation.Time,
+			CostBreakdown: estimation.CostBreakdown,
+		}
+	}
+
+	assetScanEstimation.State.LastTransitionTime = utils.PointerTo(time.Now())
+
+	assetScanEstimationPatch := models.AssetScanEstimation{
+		State:      assetScanEstimation.State,
+		Estimation: assetScanEstimation.Estimation,
+	}
+	err = w.backend.PatchAssetScanEstimation(ctx, assetScanEstimationPatch, assetScanEstimationID)
+	if err != nil {
+		return fmt.Errorf("failed to update AssetScanEstimation. AssetScanEstimation=%s: %w", assetScanEstimationID, err)
+	}
+	return nil
+}
+
+// nolint:cyclop
+func (w *Watcher) reconcileAborted(ctx context.Context, assetScanEstimation *models.AssetScanEstimation) error {
+	logger := log.GetLoggerFromContextOrDiscard(ctx)
+
+	assetScanEstimationID, ok := assetScanEstimation.GetID()
+	if !ok {
+		return errors.New("invalid AssetScanEstimation: ID is nil")
+	}
+
+	// Check if AssetScanEstimation is in aborted state for more time than the timeout allows
+	if assetScanEstimation.State == nil || assetScanEstimation.State.State == nil {
+		return errors.New("invalid AssetScanEstimation: State is nil")
+	}
+
+	var transitionTimeToAbort time.Time
+	if assetScanEstimation.State.LastTransitionTime != nil {
+		transitionTimeToAbort = *assetScanEstimation.State.LastTransitionTime
+		logger.Debugf("AssetScanEstimation moved to aborted state: %s", transitionTimeToAbort)
+	}
+
+	now := time.Now()
+	abortTimedOut := now.After(transitionTimeToAbort.Add(w.abortTimeout))
+	if !abortTimedOut {
+		logger.Tracef("AssetScanEstimation in aborted state is not timed out yet. TransitionTime=%s Timeout=%s",
+			transitionTimeToAbort, w.abortTimeout)
+		return nil
+	}
+	logger.Tracef("AssetScanEstimation in aborted state is timed out. TransitionTime=%s Timeout=%s",
+		transitionTimeToAbort, w.abortTimeout)
```

Comment:
We don't need an abort timeout for AssetScanEstimation, we have this in AssetScan because we wait for the CLI Scanner to process aborted. For AssetScanEstimations we just need to process it instantly because its all happening in this flow.